### ISSUE TITLE
Topology changed after twice recovery.

### ISSUE THREAD

**Issue by bluven** [presslabs/mysql-operator#309]
I' was trying to figure out how does recovery work. After two successive recoveries, the topology changed.

![Screenshot from 2019-04-29 16-29-51](https://user-images.githubusercontent.com/1582621/56884990-eff91900-6a9d-11e9-8311-cabc5b4baa93.png)

Is this supposed to happen?

I triggerd recovery by changing mysqlConf in cluster spec. After the first recovery, mc-mysql-1 is selected as the new master, but during the second recovery process,  mc-mysql-1 is down and mc-mysql-0 is selected as new master again, but mc-mysql-2 is still replicating from mc-mysql-1. Then mc-mysql-0 is down. I noticed at that time there was warning say something like: cluster XXX is forbidden from recovery due to previous recovery.

mc-mysql-0 also has replication setting, although no IO_THREAD and SQL_THREAD: 
```
show slave status;
*************************** 1. row ***************************
               Slave_IO_State: 
                  Master_Host: //mc-mysql-1.mc-mysql-nodes.default
                  Master_User: repl_Y5Nxt
                  Master_Port: 3306
                Connect_Retry: 10
              Master_Log_File: 
          Read_Master_Log_Pos: 4
               Relay_Log_File: mc-mysql-0-relay-bin.000001
                Relay_Log_Pos: 4
        Relay_Master_Log_File: 
             Slave_IO_Running: No
            Slave_SQL_Running: No
              Replicate_Do_DB: 
          Replicate_Ignore_DB: 
           Replicate_Do_Table: 
       Replicate_Ignore_Table: 
      Replicate_Wild_Do_Table: 
  Replicate_Wild_Ignore_Table: 
                   Last_Errno: 0
                   Last_Error: 
                 Skip_Counter: 0
          Exec_Master_Log_Pos: 0
              Relay_Log_Space: 0
              Until_Condition: None
               Until_Log_File: 
                Until_Log_Pos: 0
           Master_SSL_Allowed: No
           Master_SSL_CA_File: 
           Master_SSL_CA_Path: 
              Master_SSL_Cert: 
            Master_SSL_Cipher: 
               Master_SSL_Key: 
        Seconds_Behind_Master: NULL
Master_SSL_Verify_Server_Cert: No
                Last_IO_Errno: 0
                Last_IO_Error: 
               Last_SQL_Errno: 0
               Last_SQL_Error: 
  Replicate_Ignore_Server_Ids: 
             Master_Server_Id: 0
                  Master_UUID: 
             Master_Info_File: mysql.slave_master_info
                    SQL_Delay: 0
          SQL_Remaining_Delay: NULL
      Slave_SQL_Running_State: 
           Master_Retry_Count: 86400
                  Master_Bind: 
      Last_IO_Error_Timestamp: 
     Last_SQL_Error_Timestamp: 
               Master_SSL_Crl: 
           Master_SSL_Crlpath: 
           Retrieved_Gtid_Set: 
            Executed_Gtid_Set: 5627ede4-6a4b-11e9-b897-12ee355a59a0:1-4297,
c276c0ce-6a4b-11e9-9f84-3a982af44778:1-2944
                Auto_Position: 1
         Replicate_Rewrite_DB: 
                 Channel_Name: 
           Master_TLS_Version: 

```

I found this accidentally, I'll do more tests.

**Comment by AMecea** [comment]
Hi @bluven, thank you for opening this issue. 

Orchestrator, once a cluster is recovered, issues a `pending recovery` and until that is acknowledged the orchestrator does not perform any failover. The operator acknowledges a recovery after 10 minutes after the cluster was ready.

I will do more tests, too.

**Comment by bluven** [comment]
@AMecea I did another test and it happened again at first recovery.

**Comment by AMecea** [comment]
This is fixed in the latest version `v0.3.0`. 

@bluven can you please try again and check if this happens again?
